### PR TITLE
feat: add Breadcrumbs component

### PR DESCRIPTION
**Pull request by christian-dinh**
### Overview

[Ported from portal-app](https://github.com/codecademy-engineering/portal-app/tree/develop/src/components/Breadcrumbs), as seen on `resources/docs` and `resources/videos`

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [REACH-1078](https://codecademy.atlassian.net/browse/REACH-1078)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.f404cb.0<br/>@codecademy/gamut-kit@0.6.38-alpha.f404cb.0<br/>@codecademy/gamut-labs@28.7.0-alpha.f404cb.0<br/>@codecademy/gamut@42.3.0-alpha.f404cb.0<br/>@codecademy/monaco@0.5.28-alpha.f404cb.0<br/>@codecademy/styleguide@52.5.0-alpha.f404cb.0<br/></pre>

**Review by bndiep — APPROVED** [review]
Documentation makes sense to me. Looks good!

**Review by katiezutter — CHANGES_REQUESTED** [review]
So funny we're working on this at the same time! Here's my PR to fix an accessibility issue with the breadcrumbs: https://github.com/codecademy-engineering/portal-app/pull/1259

Would you mind transferring over the changes I made here? https://github.com/codecademy-engineering/portal-app/pull/1259/files#diff-b621e2a442c16e32861f41859fd65dad94b266e1acdc590efe3ff9e8fba67af7L7

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.85f38f.0<br/>@codecademy/gamut-kit@0.6.38-alpha.85f38f.0<br/>@codecademy/gamut-labs@28.7.0-alpha.85f38f.0<br/>@codecademy/gamut@42.3.0-alpha.85f38f.0<br/>@codecademy/monaco@0.5.28-alpha.85f38f.0<br/>@codecademy/styleguide@52.5.0-alpha.85f38f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d77d63354cdc2c35374f2d--gamut-preview.netlify.app>https://61d77d63354cdc2c35374f2d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d77d63354cdc2c35374f2d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.d7a2a1.0<br/>@codecademy/gamut-kit@0.6.38-alpha.d7a2a1.0<br/>@codecademy/gamut-labs@28.7.0-alpha.d7a2a1.0<br/>@codecademy/gamut@42.3.0-alpha.d7a2a1.0<br/>@codecademy/monaco@0.5.28-alpha.d7a2a1.0<br/>@codecademy/styleguide@52.5.0-alpha.d7a2a1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d8793a48a39c0082746d92--gamut-preview.netlify.app>https://61d8793a48a39c0082746d92--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d8793a48a39c0082746d92>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.39-alpha.f08ee7.0<br/>@codecademy/gamut-labs@28.7.0-alpha.f08ee7.0<br/>@codecademy/styleguide@52.6.0-alpha.f08ee7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d89bb7ab867c231b27696a--gamut-preview.netlify.app>https://61d89bb7ab867c231b27696a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d89bb7ab867c231b27696a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.40-alpha.749174.0<br/>@codecademy/gamut-labs@28.7.0-alpha.749174.0<br/>@codecademy/styleguide@52.6.0-alpha.749174.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61df4065cdc24e3e5b327ec7--gamut-preview.netlify.app>https://61df4065cdc24e3e5b327ec7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61df4065cdc24e3e5b327ec7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.42-alpha.12e334.0<br/>@codecademy/gamut-labs@28.7.0-alpha.12e334.0<br/>@codecademy/styleguide@53.1.0-alpha.12e334.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.43-alpha.859de1.0<br/>@codecademy/gamut-labs@28.7.0-alpha.859de1.0<br/>@codecademy/styleguide@53.1.0-alpha.859de1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61eae5533f14c136d3cfb3ae--gamut-preview.netlify.app>https://61eae5533f14c136d3cfb3ae--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61eae5533f14c136d3cfb3ae>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61eae56c53d5382c16d11aae--gamut-preview.netlify.app>https://61eae56c53d5382c16d11aae--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61eae56c53d5382c16d11aae>Deploy Logs</a></p>

**Review by katiezutter — APPROVED** [review]
Looks great! Thanks for transferring this over! 🎉

File: packages/gamut-labs/src/Breadcrumbs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+const BreadcrumbAnchor = styled(Anchor)`
+  white-space: nowrap;
+  overflow: hidden;
+  width: 100%;
+  display: inline;
+`;
+
+const BreadcrumbPart = styled(Box)`
+  list-style-type: none;
```
Comment (dreamwasp) [inline packages/gamut-labs/src/Breadcrumbs/index.tsx:13]: swapping these over to use the gamut [css](https://gamut.codecademy.com/?path=/docs/foundations-system-variants--page#css) method is a quick fix + will give ya more consistent css ordering as well as has some perf bonuses

**Review by dreamwasp — APPROVED** [review]
tiniest nit but looks great to me!!

File: packages/gamut-labs/src/Breadcrumbs/index.tsx (hunk 1 above)
Comment (christian-dinh) [reply packages/gamut-labs/src/Breadcrumbs/index.tsx:13]: oh nice, ok I got them mostly swapped over but I'm not sure what to do about the styled component selector:
```
&:first-of-type ${BreadcrumbAnchor} {
  ...
}
```

tried this but typescript hates it:
```
[`&:first-of-type ${BreadcrumbAnchor}`]: {
  ...
}
```

Comment (christian-dinh) [reply packages/gamut-labs/src/Breadcrumbs/index.tsx:13]: I guess I could just change `BreadCrumbAnchor` to  `a` actually

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.43-alpha.22c1e7.0<br/>@codecademy/gamut-labs@28.7.0-alpha.22c1e7.0<br/>@codecademy/styleguide@53.1.0-alpha.22c1e7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61eb10fd71d12e6d6e434c98--gamut-preview.netlify.app>https://61eb10fd71d12e6d6e434c98--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61eb10fd71d12e6d6e434c98>Deploy Logs</a></p>

**Pull request merged**